packages/@aws-cdk/sdk-v2-to-v3-adapter/lib/api-call.ts

```diff
@@ -0,0 +1,249 @@
+import type { AwsCredentialIdentityProvider } from '@smithy/types';
+import { coerceApiParameters } from './coerce-api-parameters';
+import { findV3ClientConstructor } from './find-client-constructor';
+import { normalizeActionName, normalizeServiceName } from './sdk-info';
+
+export interface InvokeOptions {
+  /**
+   * The SDKv3 package for the service.
+   *
+   * @default - Load the package automatically
+   */
+  readonly sdkPackage?: any;
+
+  /**
+   * Override API version
+   *
+   * @default - Use default API version
+   */
+  readonly apiVersion?: string;
+
+  /**
+   * Override region
+   *
+   * @default - Current region
+   */
+  readonly region?: string;
+
+  /**
+   * Override credentials
+   *
+   * @default - Default credentials
+   */
+  readonly credentials?: AwsCredentialIdentityProvider;
+
+  /**
+   * Parameters to the API call
+   */
+  readonly parameters: Record<string, unknown>;
+
+  /**
+   * Flatten the response object
+   *
+   * Instead of a nested object structure, return a map of `{ string -> value }`, with the keys
+   * being the paths to each primitive value.
+   *
+   * @default false
+   */
+  readonly flattenResponse?: boolean;
+}
+
+/**
+ * Wrapper to make an SDKv3 API call, with SDKv2 compatibility
+ */
+export class ApiCall {
```

Comment:
I wonder if we should name this coupling it to the version of the sdk it uses? IE `V3ApiCall`. I get that it's an API call and not necessarily an "SdkCall", but since `v3PackageName` is a public field it seems to be coupled to v3 of the sdk and once v4 is a thing we would essentially deprecate it in favor of a new one?

Reply:
Or is this only for testing purposes too?

Reply:
With a little luck for V4 we would keep the public API the same... and yeah maybe add a `v4PackageName` and/or remove the old one.

Reply:
Honestly, the only reason the `v3PackageName` is in there is because I didn't want to lift `installLatestAwsSdk` into this as well (which I friggin' hate 🤪), and so it needs to be POSSIBLE for someone else to load the SDK from somewhere